app/controllers/claims/schools/mentors_controller.rb

```diff
@@ -42,10 +45,18 @@ def set_mentor
     @mentor = @school.mentors.find(params.require(:id))
   end
 
+  def set_mentor_membership
+    @mentor_membership = @mentor.mentor_memberships.find_by!(school: @school)
+  end
+
   def authorize_mentor
     authorize @mentor || Claims::Mentor
   end
 
+  def authorize_mentor_membership
+    authorize @mentor_membership || Claims::MentorMembership
```

Comment:
The `|| Claims::MentorMembership` is only needed when we want to authorise collections of mentor memberships. We don't need to use this for this case.

```ruby
authorize @mentor_membership
```

is sufficient.